Bots/NbaBot.cs

```diff
@@ -48,37 +79,123 @@ protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivi
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A unit of execution.</returns>
-        protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersAdded, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        protected override async Task OnConversationUpdateActivityAsync(
+            ITurnContext<IConversationUpdateActivity> turnContext,
+            CancellationToken cancellationToken)
         {
-            if (membersAdded is null)
+            if (turnContext is null)
             {
-                throw new ArgumentNullException(nameof(membersAdded));
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            try
+            {
+                var activity = turnContext.Activity;
+
+                this.telemetryClient.TrackTrace($"Received conversationUpdate activity");
+                this.telemetryClient.TrackTrace($"conversationType: {activity.Conversation.ConversationType}, membersAdded: {activity.MembersAdded?.Count}, membersRemoved: {activity.MembersRemoved?.Count}");
+
+                await this.OnMembersAddedAsync(activity.MembersAdded, turnContext, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                this.telemetryClient.TrackTrace($"Error processing conversationUpdate: {ex.Message}", SeverityLevel.Error);
+                this.telemetryClient.TrackException(ex);
+                throw;
             }
+        }
 
+        /// <summary>
+        /// This method fires whenever a new member has been added.
+        /// </summary>
+        /// <param name="membersAdded">The list of members being added.</param>
+        /// <param name="turnContext">The current turn/execution flow.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A unit of execution.</returns>
+        protected override async Task OnMembersAddedAsync(
+            IList<ChannelAccount> membersAdded,
+            ITurnContext<IConversationUpdateActivity> turnContext,
+            CancellationToken cancellationToken)
+        {
             if (turnContext is null)
             {
                 throw new ArgumentNullException(nameof(turnContext));
             }
 
-            var welcomeText = "Hello and welcome!";
+            if (membersAdded is null)
+            {
+                throw new ArgumentNullException(nameof(membersAdded));
+            }
+
+            this.telemetryClient.TrackTrace("Sending the welcome message");
+            var welcomeText = BotResource.WelcomeText;
             foreach (var member in membersAdded)
             {
                 if (member.Id != turnContext.Activity.Recipient.Id)
                 {
-                    await turnContext.SendActivityAsync(MessageFactory.Text(welcomeText, welcomeText), cancellationToken).ConfigureAwait(false);
```

Comment:
You don't have to have two strings for welcomeText, the second one is an optional string.

Reply:
This was part of the original EchoBot sample again